crates/biome_js_analyze/src/lint/nursery/use_google_font_display.rs

```diff
@@ -0,0 +1,124 @@
+    ///
+    /// The `font-display` property controls how a font is displayed while it is loading. When using Google Fonts,
+    /// it's important to specify an appropriate value for this property to ensure good user experience and prevent layout shifts.
+    ///
+    /// This rule flags the absence of the `font-display` parameter, or the usage of less optimal values such as `auto`, `block`, or `fallback`.
+    /// Using `&display=optional` is generally recommended as it minimizes the risk of invisible text or layout shifts.
+    /// In cases where swapping to the custom font after it has loaded is important, consider using `&display=swap`.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One" />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One&display=auto" />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One&display=block" />
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One&display=fallback" />
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <link href="https://fonts.googleapis.com/css2?family=Krona+One&display=optional" rel="stylesheet" />
+    /// ```
+    ///
+    /// ```jsx
+    /// <link href="https://fonts.googleapis.com/css2?display=unknown" rel="stylesheet" />
+    /// ```
+    ///
+    /// ```jsx
+    /// <link rel="stylesheet" />
+    /// ```
+    pub UseGoogleFontDisplay {
+        version: "next",
+        name: "useGoogleFontDisplay",
+        language: "jsx",
+        sources: &[RuleSource::EslintNext("google-font-display")],
+        source_kind: RuleSourceKind::SameLogic,
+        recommended: false,
+    }
+}
+
+impl Rule for UseGoogleFontDisplay {
+    type Query = Ast<AnyJsxElement>;
+    type State = (String, TextRange);
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let element = ctx.query();
+
+        if element.name().ok()?.name_value_token()?.text_trimmed() != "link" {
+            return None;
+        }
+
+        let href = element.find_attribute_by_name("href");
+        let initializer = href?.initializer()?.value().ok()?;
+        let jsx_string = initializer.as_jsx_string();
+        let href_text = jsx_string?.inner_string_text().ok()?;
+        let is_google_font = href_text.starts_with("https://fonts.googleapis.com/css");
+
+        if is_google_font {
+            return Some((
+                href_text.text().to_string(),
+                jsx_string?.value_token().ok()?.text_trimmed_range(),
+            ));
+        }
+
+        None
+    }
+
+    fn diagnostic(_: &RuleContext<Self>, (href, range): &Self::State) -> Option<RuleDiagnostic> {
```

Comment:
Inside the diagnostic you have logic that might not print a message. If that happens, it means that this logic should be moved inside `run`